### PR TITLE
DOC: Update docs to reflect that Index can hold int64, int32 etc. arrays

### PR DESCRIPTION
**Pull request by topper-123**
Add whatsnew and make doc changes in order to document that `Index` can hold int64, int32 etc. arrays. Removes references to `Int64Index` etc. from the docs.

xref #42717.

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -28,6 +28,79 @@ The available extras, found in the :ref:`installation guide<install.dependencies
...
+   dtype ``int64``, while they were ``int32`` for :class:`DatetimeArray`. They are now
+   ``int32`` on ``DateTimeIndex`` also:
+
+   .. ipython:: python
+
+       idx = pd.date_range(start='1/1/2018', periods=3, freq='M')
+       idx.array.year
+       idx.year
+
+3. Level dtypes on Indexes from :attr:`Series.sparse.from_coo` are now of dtype ``int32``.
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:84]:
```suggestion
3. Level dtypes on Indexes from :meth:`Series.sparse.from_coo` are now of dtype ``int32``.
```

Would be good to state this matches scipy and used to be forced to int64

Comment (topper-123) [reply doc/source/whatsnew/v2.0.0.rst]: I added this to the latest commit.

**Comment by topper-123** [comment]
I think the ARM failure is unrelated.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Nice whatsnew summary!

**Pull request merged**